In [1]:
import pandas as pd

In [2]:
#Get information of Sale file (df1)
df1 = pd.read_csv('shipped_order.csv')
df1

Order No Order Date  Line No  Ship To Address No  Item Code  Order Qty  \
0        528758   1/3/2019        1           1358538.0     111931       70.0   
1        528791   1/3/2019        1           1254798.0    1029071       10.0   
2        528791   1/3/2019        2           1254798.0    1033341       10.0   
3        528791   1/3/2019        3           1254798.0    1040827        5.0   
4        528791   1/3/2019        4           1254798.0   10106111       10.0   
...         ...        ...      ...                 ...        ...        ...   
28310    543761  2/28/2019        1          43825307.0     113106      380.0   
28311    543762  2/28/2019        1           1255123.0     113106      300.0   
28312    543764  2/28/2019        2          43825307.0   10131275      400.0   
28313    543765  2/28/2019        1          43825307.0   10131275      300.0   
28314    543766  2/28/2019        2           1255123.0   10131275      100.0   

       Quantity Shipped  
0                  70.0  
1                  10.0  
2                  10.0  
3                   5.0  
4                  10.0  
...                 ...  
28310             380.0  
28311             300.0  
28312             400.0  
28313             300.0  
28314             100.0  

[28315 rows x 7 columns]

In [3]:
df1.isna().any()

Order No              False
Order Date            False
Line No               False
Ship To Address No    False
Item Code             False
Order Qty             False
Quantity Shipped      False
dtype: bool

In [4]:
df1.duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
28310    False
28311    False
28312    False
28313    False
28314    False
Length: 28315, dtype: bool

In [5]:
print(len(set(df1['Order No'])))

9411


In [6]:
print(len(set(df1['Ship To Address No'])))

708


In [7]:
print(len(set(df1['Item Code'])))

64


In [8]:
print(len(set(df1['Line No'])))

47


In [9]:
df1.dtypes

Order No                int64
Order Date             object
Line No                 int64
Ship To Address No    float64
Item Code               int64
Order Qty             float64
Quantity Shipped      float64
dtype: object

In [10]:
#Concert to datetime for order date
df1['Order Date'] = pd.to_datetime(df1['Order Date'])
df1.dtypes

Order No                       int64
Order Date            datetime64[ns]
Line No                        int64
Ship To Address No           float64
Item Code                      int64
Order Qty                    float64
Quantity Shipped             float64
dtype: object

In [11]:
#Get information of Cancel file (df2)
df2 = pd.read_csv('canceled_order.csv')
df2

Order Number  Line Number Order Date  Ship To Address No  Item Code  \
0           528703            1   1/3/2019           1857566.0   10135139   
1           528705            1   1/3/2019           1857566.0   10135140   
2           528706            2   1/3/2019           1857566.0   10135138   
3           528707            1   1/3/2019           1857566.0   10135132   
4           528708            1   1/3/2019           1857566.0   10135359   
...            ...          ...        ...                 ...        ...   
3657        543485            8  2/25/2019           2203715.0   10135140   
3658        543488            6  2/25/2019           2203449.0    1040468   
3659        543488            7  2/25/2019           2203449.0    1040466   
3660        543491            5  2/25/2019           2203449.0   10135358   
3661        543491            9  2/25/2019           2203449.0   10135363   

      Quantity Ordered  Quantity Canceled  
0                    1                  1  
1                    1                  1  
2                    1                  1  
3                    1                  1  
4                    1                  1  
...                ...                ...  
3657                 3                  3  
3658                 2                  2  
3659                 1                  1  
3660                21                 21  
3661                 6                  6  

[3662 rows x 7 columns]

In [12]:
df2.dtypes

Order Number            int64
Line Number             int64
Order Date             object
Ship To Address No    float64
Item Code               int64
Quantity Ordered        int64
Quantity Canceled       int64
dtype: object

In [13]:
#Convert datetime for order date
df2['Order Date'] = pd.to_datetime(df2['Order Date'])
df2.dtypes

Order Number                   int64
Line Number                    int64
Order Date            datetime64[ns]
Ship To Address No           float64
Item Code                      int64
Quantity Ordered               int64
Quantity Canceled              int64
dtype: object

In [14]:
df2.isna().any()

Order Number          False
Line Number           False
Order Date            False
Ship To Address No    False
Item Code             False
Quantity Ordered      False
Quantity Canceled     False
dtype: bool

In [15]:
df2.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
3657    False
3658    False
3659    False
3660    False
3661    False
Length: 3662, dtype: bool

In [16]:
print(len(set(df2['Order Number'])))

1925


In [17]:
print(len(set(df2['Line Number'])))

27


In [18]:
print(len(set(df2['Ship To Address No'])))

605


In [19]:
print(len(set(df2['Item Code'])))

73


#Conclusion for sale file: 
- Sale file have 7 columns, 28 315 rows 
- Data clean - no NaN, do duplicate 
- Data type: done convert to datetime 
- Data set
        47 Line 
        64 Item 
        708 Address/Customer 
        9411 Order 

#Conclusion for cancel file: 
- Cancel file have 7 columns, 3662 rows 
- Data clean - no NaN, do duplicate 
- Data type: done convert to datetime. 2 last columns different types vs Sale 
- data set: 
        27 Line -> 20 line no cancel 
        73 Item -> 9 item cancel rate = 100% 
        605 Address/Customer 
        1925 Order

Analysis Purpose: Get initial understanding of customer Question:

Daily/Period volume. Sale rate
Their success rate/cancel rate
Classify into categories of SKU criteria: %delivered/order and find which way to categorize SKU/Customer 

In [20]:
df1.columns

Index(['Order No', 'Order Date', 'Line No', 'Ship To Address No', 'Item Code',
       'Order Qty', 'Quantity Shipped'],
      dtype='object')

In [21]:
df2.columns

Index(['Order Number', 'Line Number', 'Order Date', 'Ship To Address No',
       'Item Code', 'Quantity Ordered', 'Quantity Canceled'],
      dtype='object')